Adding coding style checker like eslint?

Here is example.
```js
// toLint.js
1  console.log("Hello World");
2    var hi = "HHIHI";
3
4  function ji(){
5    return 123;
6  }
```

By using `eslint`, the example set style to `google`

```sh
$ eslint index.js

/tmp/index.js
  1:13  error  Strings must use singlequote              quotes
  2:3   error  Unexpected var, use let or const instead  no-var
  2:7   error  'hi' is assigned a value but never used   no-unused-vars
  2:12  error  Strings must use singlequote              quotes
  4:1   error  Missing JSDoc comment                     require-jsdoc
  4:10  error  'ji' is defined but never used            no-unused-vars
  4:14  error  Missing space before opening brace        space-before-blocks
```

- **Q1**: Do we need this?
- **Q2**: What style should be used?
  - airbnb
  - google
  - standard

Any suggestion?

--- a/.eslintignore
+++ b/.eslintignore
@@ -1,0 +1,1 @@
+crypto.js

--- a/.eslintrc.json
+++ b/.eslintrc.json
@@ -1,0 +1,21 @@
+{
+  "env": {
+    "commonjs": true,
+    "node": true
+  },
+  "parserOptions": {
+    "ecmaVersion": 2015
+  },
+  "rules": {
+    "space-before-function-paren": [
+      "error",
+      {
+        "anonymous": "always",
+        "named": "never",
+        "asyncArrow": "always"
+      }
+    ],
+    "no-console": "off"
+  },
+  "extends": "eslint:recommended"
+}

--- a/.travis.yml
+++ b/.travis.yml
@@ -17,6 +17,6 @@ jobs:
       env:
         - test: Editor Config check
       install:
-      - npm i -g echint
+      - npm i
       script:
-      - echint -i "sri/**/*" -i "public/**/*" -v "tutorials/**/*" -v
+      - npm test

--- a/package.json
+++ b/package.json
@@ -4,7 +4,9 @@
   "description": "new-website ===========",
   "main": "server.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
+    "test": "npm run test:echint && npm run test:eslint",
+    "test:echint": "echint -i \"sri/**/*\" -i \"public/**/*\" -v \"tutorials/**/*\" -v",
+    "test:eslint": "eslint *.js",
     "start:api": "node --expose-gc apiServer.js",
     "start:web": "node --expose-gc webServer.js",
     "dev:api": "LOCAL=true node apiServer.js",
@@ -45,5 +47,9 @@
     "mustache": "^2.3.2",
     "newline-remove": "^1.0.2",
     "throng": "^4.0.0"
+  },
+  "devDependencies": {
+    "echint": "^4.0.2",
+    "eslint": "^5.16.0"
   }
 }

--- a/reindex.js
+++ b/reindex.js
@@ -56,7 +56,7 @@ function load(next) {
 
     var SRI;
     try {
-      json = JSON.parse(fs.readFileSync('sri/' + library.name + '/' + library.version + '.json', "utf8"));
+      var json = JSON.parse(fs.readFileSync('sri/' + library.name + '/' + library.version + '.json', "utf8"));
       SRI = json[library.filename];
     } catch (e) {
       SRI = "";
@@ -102,7 +102,7 @@ function crawl(gnext) {
         return null;
       }
 
-      var m = url.match(/.*github\.com[\/:]([^\/]+)\/([^\/]+)(\/.*|\.git)?$/);
+      var m = url.match(/.*github\.com[/:]([^/]+)\/([^/]+)(\/.*|\.git)?$/);
       if (!m) {
         return null;
       }
@@ -178,7 +178,7 @@ function initIndex(next) {
     attributesForFaceting: ['fileType', 'keywords'],
     optionalWords: ['js', 'css'], // those words are optional (jquery.colorbox.js <=> jquery.colorbox)
     ranking: ['typo', 'words', 'proximity', 'attribute', 'custom'] // removed the "exact" criteria conflicting with the "keywords" array containing exact forms
-  }, function (error, content) {
+  }, function (error) {
     if (error) {
       printError(error);
       return;
@@ -190,7 +190,7 @@ function initIndex(next) {
 
 function push(next) {
   console.log('* Indexing ' + LIBRARIES.length + ' libraries');
-  index.addObjects(LIBRARIES, function (error, content) {
+  index.addObjects(LIBRARIES, function (error) {
     if (error) {
       printError(error);
       return;
@@ -202,7 +202,7 @@ function push(next) {
 
 function commit(next) {
   console.log('* Moving index to production');
-  client.moveIndex('libraries.tmp', 'libraries', function (error, content) {
+  client.moveIndex('libraries.tmp', 'libraries', function (error) {
     if (error) {
       printError(error);
       return;

--- a/webServer.js
+++ b/webServer.js
@@ -270,7 +270,7 @@ function start() {
 
       if (assets.gennedFileNames === undefined) {
         var fileArray = [];
-        assets.files.map(function (fileName, index) {
+        assets.files.map(function (fileName) {
           var fileExtension = path.extname(fileName);
           var fileType = fileExtension.substring(1) || 'unknown';
           fileArray.push({ name: fileName, fileType: fileType });
@@ -528,7 +528,7 @@ function start() {
     }));
   });
 
-  app.use(function (err, req, res, next) {
+  app.use(function (err, req, res) {
     console.error(err.stack);
     res.status(500).send(generatePage({
       reqUrl: req.url,

